hyperspy/signal.py

```diff
@@ -3199,16 +3199,18 @@ def map(self, function,
             parallel=None, **kwargs):
         """Apply a function to the signal data at all the coordinates.
 
-        The function must operate on numpy arrays and the output *must have the
-        same dimensions as the input*. The function is applied to the data at
+        The function must operate on numpy arrays, but the output no longer has
```

Comment:
I would cut it short: "The function must operate on numpy arrays." as new users are not interested in the history of the function.